### PR TITLE
Upgrade python tests framework

### PR DESCRIPTION
**Pull request by pik694**
The following PR introduces set of changes that upgrade python testing framework.
The RLP encoding library has been upgraded.
The EVM has been changed from deprecated `pyethreum` to `ganache`.

This is  a part of changes required to solve #150.

References #204

### CODE REVIEW

File: tests/conveniece_wrappers.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        return tx_hash
+
+    def _get_next_timestamp(self):
+        next_timestamp = self._next_timestamp or (self.getBlock('latest')['timestamp'] + 1)
+        self._next_timestamp = None
+        return next_timestamp
+
+
+class ConvenienceContractWrapper:
+    default_params = {'gasPrice': 0, 'gas': 4 * 10 ** 6}
```
Comment (paulperegud) [inline tests/conveniece_wrappers.py]: explain why we want to bypass `ganache-cli` gas estimation functionality

File: tests/conveniece_wrappers.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
+from eth_tester.exceptions import TransactionFailed
+from web3 import eth
+from web3._utils.datatypes import PropertyCheckingFactory
+from web3.contract import Contract
+
+
+class AutominingEth(eth.Eth):
+
```
Comment (paulperegud) [inline tests/conveniece_wrappers.py]: please explain what we do here and why we do it

File: tests/contracts/root_chain/test_process_exits.py [hunk 3]
```diff
@@ -202,49 +205,60 @@ def test_processing_exits_with_specifying_top_exit_id_is_possible(testlang):
...
     testlang.process_exits(NULL_ADDRESS, testlang.get_standard_exit_id(utxo1.spend_id), 1)
     testlang.forward_timestamp(2 * MIN_EXIT_PERIOD + 1)
 
-    testlang.process_exits(NULL_ADDRESS, testlang.get_standard_exit_id(utxo2.spend_id), 3)
-    three_exits_tx_gas = testlang.ethtester.chain.last_gas_used()
-    assert three_exits_tx_gas > 3516  # value from _tx_race_short_circuit
+    tx_hash = testlang.process_exits(NULL_ADDRESS, testlang.get_standard_exit_id(utxo2.spend_id), 3)
+    tx_receipt = w3.eth.getTransactionReceipt(tx_hash)
+    three_exits_tx_gas = tx_receipt['gasUsed']
+    assert three_exits_tx_gas > 26258  # value from _tx_race_short_circuit
```
[the remark this answers is not in the document]
Comment (pik694) [reply tests/contracts/root_chain/test_process_exits.py]: The gas usage topic appears to be something that needs further investigation, therefore I created a new issue  #238.

File: plasma_core/transaction.py [hunk 4]
```diff
@@ -103,11 +99,11 @@ def encoded(self):
...
-    def sign(self, index, key, verifyingContract=None):
-        hash = hash_struct(self, verifyingContract=verifyingContract)
-        sig = sign(hash, key)
-        self.signatures[index] = sig
-        self._signers[index] = get_signer(hash, sig) if sig != NULL_SIGNATURE else NULL_ADDRESS
+    def sign(self, index, account, verifyingContract=None):
+        msg_hash = hash_struct(self, verifyingContract=verifyingContract)
+        sig = account.key.sign_msg_hash(msg_hash)
+        self.signatures[index] = sig.to_bytes()
+        self._signers[index] = sig.recover_public_key_from_msg_hash(msg_hash).to_canonical_address() if sig != NULL_SIGNATURE else NULL_ADDRESS
```
[the remark this answers is not in the document]
Comment (pik694) [reply plasma_core/transaction.py]: Added a comment to `conftest.py` file.

File: tests/contracts/root_chain/test_process_exits.py [hunk 5]
```diff
@@ -202,49 +205,60 @@ def test_processing_exits_with_specifying_top_exit_id_is_possible(testlang):
...
-    with pytest.raises(TransactionFailed):
-        testlang.process_exits(NULL_ADDRESS, testlang.get_standard_exit_id(utxo1.spend_id), 3)
-    short_circuit_gas = testlang.ethtester.chain.last_gas_used()
-    assert short_circuit_gas < 67291  # value from _tx_race_normal
 
+    w3.eth.disable_auto_mine()
+
+    tx_hash = root_chain.functions\
+        .processExits(NULL_ADDRESS, testlang.get_standard_exit_id(utxo1.spend_id), 3)\
+        .transact({'gas': 100_000})  # reasonably high amount of gas (otherwise it fails on gas estimation)
```
[the remark this answers is not in the document]
Comment (pik694) [reply tests/contracts/root_chain/test_process_exits.py]: The problem is that `ganache-cli` panics at `gasEstimate` RPC call, when the transaction that is being estimated fails.

File: tests/contracts/root_chain/test_long_run.py [hunk 6]
```diff
@@ -10,11 +10,12 @@ def subtract(left, right):
...
 def pick(left):
     item = random.choice(left)
-    return (subtract(left, [item]), item)
+    return subtract(left, [item]), item
 
 
-@pytest.mark.slow()
-def test_slow(testlang):
+# @pytest.mark.slow()
+@pytest.mark.skip("In this model it takes forever")
```
[the remark this answers is not in the document]
Comment (pik694) [reply tests/contracts/root_chain/test_long_run.py]: I have added a job that runs once every Monday at midnight - I made a quick test and it should  take ~ 6-7 hours.

**Pull request merged**